tests/test_types.py

```diff
@@ -106,10 +106,37 @@ def test_parse_type_no_basic_types():
     nose.tools.assert_true(byte.size, 8)
     nose.tools.assert_false(byte.signed)
 
+def test_self_referential_struct_or_union():
+    struct_llist = angr.types.parse_type('struct llist { int data; struct llist *next; }')
+    next_struct_llist = struct_llist.fields['next'].pts_to
+    nose.tools.assert_true(len(next_struct_llist.fields) == 2)
+    nose.tools.assert_is_instance(next_struct_llist.fields['data'], SimTypeInt)
+    nose.tools.assert_is_instance(next_struct_llist.fields['next'], SimTypePointer)
+
+    union_heap = angr.types.parse_type('union heap { int data; union heap *forward; }')
+    forward_union_heap = union_heap.members['forward'].pts_to
+    nose.tools.assert_true(len(forward_union_heap.members) == 2)
+    nose.tools.assert_is_instance(forward_union_heap.members['data'], SimTypeInt)
+    nose.tools.assert_is_instance(forward_union_heap.members['forward'], SimTypePointer)
+
+def test_avoid_modify_stored_union_struct_while_parsing():
+    union_a = angr.types.parse_type('union a')
+    angr.types.register_types(union_a)
+    angr.types.parse_type('union a { int x; }')
+    nose.tools.assert_true(len(union_a.members) == 0)
```

Comment:
use assert_equals instead of assert_true

Reply:
I've just corrected this.